libapicore/ApiServer.cpp

```diff
@@ -898,3 +914,143 @@ Json::Value ApiConnection::getMinerStatHR()
+    jshares["rejected"] = s.getRejects(index);
+    jshares["invalid"] = s.getFailures(index);
+    jshares["acceptedstale"] = s.getAcceptedStales(index);
+    auto solution_lastupdated = std::chrono::duration_cast<std::chrono::minutes>(
+        std::chrono::steady_clock::now() - s.getLastUpdated(index));
+    jshares["lastupdate"] = uint64_t(solution_lastupdated.count());
+    jRes["shares"] = jshares;
+
+
+    /* Hardware */
+    if (index < p.minerMonitors.size())  // TODO: should we export 0 if we've no values ?
+    {
+        jRes["temp"] = p.minerMonitors[index].tempC;
+        jRes["fan"] = p.minerMonitors[index].fanP;
+        jRes["power"] = p.minerMonitors[index].powerW;
+    }
+    // TODO: PCI ID, Name, ... (some more infos - see listDevices())
+
+    /* Pause infos */
+    if (miner)
+        if (index < p.miningIsPaused.size())
+        {
+            MinigPauseReason pause_reason = miner->get_mining_paused();
+            MiningPause m;
+            jRes["ispaused"] = m.is_mining_paused(pause_reason);
+            jRes["pause_reason"] = m.get_mining_paused_string(pause_reason);
+        }
+
+    /* Nonce infos */
+    auto segment_width = m_farm.get_segment_width();
+    uint64_t gpustartnonce =
+        m_farm.get_nonce_scrambler() + ((uint64_t)pow(2, m_farm.get_segment_width()) * index);
+    jRes["nonce_start"] = gpustartnonce;
+    jRes["nonce_stop"] = uint64_t(gpustartnonce + (uint64_t)(pow(2, segment_width)));
+
+    return jRes;
+}
+
+
+/**
+ * @brief Return a total and per GPU detailed list of current status
+ * As we return here difficulty and share counts (which are not getting resetted if we
+ * switch pool) the results may "lie".
+ * Eg: Calculating runtime, (current) difficulty and submitted shares must not match the hashrate.
+ * Inspired by Andrea Lanfranchi comment on issue 1232:
+ *    https://github.com/ethereum-mining/ethminer/pull/1232#discussion_r193995891
+ * @return The json result
+ */
+Json::Value ApiConnection::getMinerStatDetail()
+{
+/* TODO:
+   Should we always retur all values or just those we got ?
+   For API clients it can be painfull to check each item for existance
+*/
+    auto runningTime = std::chrono::duration_cast<std::chrono::minutes>(
+        std::chrono::steady_clock::now() - m_farm.farmLaunched());
+
+    SolutionStats s = m_farm.getSolutionStats();
+    WorkingProgress p = m_farm.miningProgress();
+
+    // ostringstream version;
+    Json::Value gpus;
+    Json::Value jRes;
+
+    jRes["version"] = ethminer_get_buildinfo()->project_name_with_version;  // miner version.
+    jRes["runtime"] = uint64_t(runningTime.count());  // running time, in minutes.
+
+    {
+        // Even the client should know which host was queried
+        char hostName[HOST_NAME_MAX + 1];
+        if (!gethostname(hostName, HOST_NAME_MAX + 1))
+            jRes["hostname"] = hostName;
+    }
+
+#if 0
+    ostringstream poolAddresses;
+    poolAddresses << m_farm.get_pool_addresses();
```

Comment:
As I understand you've exposed method to get active connection info directly from poolmanager.

Can we remove the redundacy of setting set/get_pool_addresses() on Farm ?